src/Microsoft.AspNet.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -94,7 +94,7 @@ public static bool TryGet(IDictionary<string, StringValues> headers, string name
                 value = values[0];
                 return true;
             }
-            value = String.Join(",", values);
+            value = string.Join(",", values.ToArray());
```

Comment:
Yep, boxes enumerator to `IEnumerator`; as items > 1 we know `.ToArray` will return an underlying array without creation.